src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3531,6 +3519,177 @@ private bool NotesAlreadyLoaded(IEnumerable<ExtraAnnotationViewInfo> notes)
+                        {
+                            note.PinnedNode = guid.ToString();
+                        }
+                    }
+                }
+
+                foreach (var note in notes)
+                {
+                    // Repopulate the group elements
+                    if (note.Nodes != null)
+                    {
+                        var updatedNodes = new List<string>();
+
+                        foreach (var node in note.Nodes)
+                        {
+                            // In case it's a node
+                            if (nodesGuids.TryGetValue(new Guid(node), out var guid))
+                            {
+                                updatedNodes.Add(guid.ToString());
+                            }
+                            // In case it's a group or a note
+                            else if (notesGuids.TryGetValue(node, out var id))
+                            {
+                                updatedNodes.Add(id);
+                            }
+                        }
+
+                        note.Nodes = updatedNodes;
+                    }
+                }
+            }
+
+            return notesGuids;
+        }
+
+        private Dictionary<Guid, Guid> UpdateNodeGUIDs(ref IEnumerable<NodeModel> nodes)
+        {
+            Dictionary<Guid, Guid> nodesGuids = new Dictionary<Guid, Guid>();
+
+            // Assign new GUIDs
+            foreach (var node in nodes)
+            {
+                var newGuid = Guid.NewGuid();
+                nodesGuids[node.GUID] = newGuid;
+                node.GUID = newGuid;
+            }
+
+            return nodesGuids;
+        }
+
+        private Dictionary<Guid, Guid> UpdateConnectorGUIDs(ref IEnumerable<ConnectorModel> connectors)
+        {
+            Dictionary<Guid, Guid> connectorsGuids = new Dictionary<Guid, Guid>();
+
+            foreach (var connector in connectors)
+            {
+                var newGuid = Guid.NewGuid();
+                connectorsGuids[connector.GUID] = newGuid;
+                connector.GUID = newGuid;
+
+                // Assign new pin GUIDs
+                foreach (var pin in connector.ConnectorPinModels)
+                {
+                    newGuid = Guid.NewGuid();
+                    pin.GUID = newGuid;
+                }
+            }
+
+            return connectorsGuids;
+        }
+
+        /// <summary>
+        /// Update GUIDs inside ExtraWorskapceViewInfo
+        /// </summary>
+        /// <param name="viewInfo">ViewInfo object containing information allowing to recreate the workspace</param>
+        /// <param name="nodesGuids">Nodes [old,new] GUID values</param>
+        /// <param name="notesGuids">Annotations [old,new] GUID values</param>
+        /// <param name="connectorsGuids">Connectors [old,new] GUID values</param>
+        private void PropagateUpdatesToViewInfo(ExtraWorkspaceViewInfo viewInfo,
```

Comment:
For this function, it seems we are only replacing Guids, I feel the function maybe can be simplified further. After some deeper look, it seems these elements do not share same interface, we cant really write utility method to avoid repeating code. So good for now.

Reply:
I was thinking the same, but they are ever so slightly different to make me not do it. They do share the same base class, but that class does not have the GUID we are after. Also, the GUID is sometimes a Guid, sometimes a string.

Reply:
All code is now reworked in favor of a much simpler method that operates directly on the json string, as discussed on Friday. It works great to on my side, but let me know if that's not ideal for whatever reason and I can update as needed.